packages/compose/src/hooks/use-instance-id/index.ts

```diff
@@ -0,0 +1,63 @@
+// Disable reason: Object and object are distinctly different types in TypeScript and we mean the lowercase object in thise case
+// but eslint wants to force us to use `Object`. See https://stackoverflow.com/questions/49464634/difference-between-object-and-object-in-typescript
+/* eslint-disable jsdoc/check-types */
+/**
+ * WordPress dependencies
+ */
+import { useMemo } from '@wordpress/element';
+
+const instanceMap = new WeakMap< object, number >();
+
+/**
+ * Creates a new id for a given object.
+ *
+ * @param {object} object Object reference to create an id for.
```

Comment:
Let's remove the types from all JSDoc comments — specifying types in TypeScript is enough

Reply:
👍 Done in [0aa7b42](https://github.com/WordPress/gutenberg/pull/43790/commits/0aa7b42b6dbe61b1c51439d73e1558f633d1cf92)